.github/workflows/preflight-release.yml

```diff
@@ -0,0 +1,69 @@
+name: Check for new Preflight release and create PR
+
+on:
+  schedule:
+    - cron: '0 8 * * *' # daily at 8 AM
+  workflow_dispatch:
+  pull_request: # TODO: to delete, only here for testing
+    branches:
+      - main
+
+jobs:
+  check-new-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Check for new release
+        id: check_release
+        run: |
+          preflight_latest=$(curl -s https://api.github.com/repos/redhat-openshift-ecosystem/openshift-preflight/releases/latest | jq -r .tag_name)
+          echo "Preflight latest release: $preflight_latest"
+          echo "preflight_latest=$preflight_latest" >> $GITHUB_OUTPUT
+
+      - name: Get current release version
+        id: get_current_release
+        run: |
+          preflight_current=$(grep 'preflight_image:' roles/preflight/defaults/main.yml | awk -F: '{print $3}' | tr -d ' "')
+          echo "Preflight current release: $preflight_current"
+          echo "preflight_current=$preflight_current" >> $GITHUB_OUTPUT
+
+      - name: Compare versions
+        id: compare_versions
+        env:
+          PREFLIGHT_LATEST: ${{ steps.check_release.outputs.preflight_latest }}
+          PREFLIGHT_CURRENT: ${{ steps.get_current_release.outputs.preflight_current }}
+        run: |
+          if [ "$PREFLIGHT_LATEST" != "$PREFLIGHT_CURRENT" ]; then
+            echo "New Preflight release found: $PREFLIGHT_LATEST, current: $PREFLIGHT_CURRENT"
+            echo "new_release=true" >> $GITHUB_OUTPUT
+          else
+            echo "Both Preflight latest and current are the same: $PREFLIGHT_LATEST"
+            echo "new_release=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Install GitHub CLI
+        if: steps.compare_versions.outputs.new_release == 'true'
+        run: sudo apt-get install -y gh
+
+      - name: Authenticate and create PR
+        if: steps.compare_versions.outputs.new_release == 'true'
+        env:
+          PREFLIGHT_LATEST: ${{ steps.check_release.outputs.preflight_latest }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Update main.yml
+          sed -i "s|quay.io/opdev/preflight:[^\"]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/defaults/main.yml
+          # Update README.md
+          sed -i "s|quay.io/opdev/preflight:[^\"]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/README.md
+          # Configure git
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          # Create a new branch, commit, and push changes
```

Comment:
you need to switch to main for testing in pull_request else you are on the branch of the PR...

Reply:
It's a bit difficult to test because we already have the latest preflight release 1.9.6 on the main branch. My test idea was:

- To introduce 1.9.5 on my branch `automate_preflight_release`, to simulate the situation with 'not latest' version 
- Automatically create new branch `preflight-release-$PREFLIGHT_LATEST` from the main branch
- Open an empty PR from `preflight-release-$PREFLIGHT_LATEST` to `main` (because both `preflight-release-$PREFLIGHT_LATEST` and `main` will contain the same latest 1.9.6)

The problem was that I was creating my new branch from the PR branch `automate_preflight_release` instead of `main`. Let me fix the issue.

Reply:
@fredericlepied could you please have a look? 

To open PR works: [GH action logs](https://productionresultssa4.blob.core.windows.net/actions-results/67d6dcb7-c5ed-401a-af39-77956770fd34/workflow-job-run-1219f5dd-843c-57d9-c582-81bc556613ac/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-06-03T10%3A30%3A03Z&sig=GvoD9dLzXCSiF40QIEnBojoirClEEG%2BFV%2FqdmNgm0lY%3D&ske=2024-06-03T20%3A38%3A03Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-06-03T08%3A38%3A03Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-06-03T10%3A19%3A58Z&sv=2023-11-03) and [PR that was opened](https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/330). It's empty because on the main branch we already use the latest Preflight.